### ISSUE TITLE
Automatically Install ghc-mod when not available locally

### ISSUE THREAD

**Issue by martinring** [flatmap/cobra#10]

**Comment by makarius** [comment]
Just a comment on ghc-mod: yesterday I prepared everything on my demonstration machine, when I discovered that its Ubuntu 14.04 LTS had an old version of ghc-mod that did not work with Cobra 1.0.1. When trying to update the installation, I trashed it accidentally and ended up spending half of the night re-installing the machine: it is now a fresh and new Ubuntu 16.04 LTS.

In Isabelle, we usually try to avoid loose dependencies like ghc-mode here. But the GHC universe is huge and complex, and I have no idea if and how it could be singled out and shrink-wrapped for a standalone distirbution of Cobra.

**Comment by martinring** [comment]
Yes, this is a problem, which is why I am investigating options on including ghc-mod automatically. I guess the way to go is to use cabal to handle everything. There seems to be an option to have an isolated cabal.

The main problem now is, that I would like some general solution as to where to store stuff like this. There seems to be no platform independent way to get a global directory for the current user, in which configuration and dependencies like these can be stored. I thought, this would just be a java environment variable or something but it looks like it is a manual process.
